src/proxy-ui-api/src/test/java/org/niis/xroad/securityserver/restapi/openapi/SystemApiControllerTest.java

```diff
@@ -335,4 +339,27 @@ public void previewAnchor() throws IOException {
         assertEquals(ANCHOR_HASH, anchor.getHash());
         assertEquals(ANCHOR_CREATED_AT, anchor.getCreatedAt().toString());
     }
+
+    @Test
+    @WithMockUser(authorities = { "VIEW_NODE_TYPE" })
+    public void getNodeTypeStandalone() {
+        ResponseEntity<NodeTypeResponse> response = systemApiController.getNodeType();
+        assertEquals(response.getBody().getNodeType(), NodeType.STANDALONE); // default value is STANDALONE
+    }
+
+    @Test
+    @WithMockUser(authorities = { "VIEW_NODE_TYPE" })
+    public void getNodeTypePrimary() {
+        when(systemService.getServerNodeType()).thenReturn(SystemProperties.NodeType.MASTER);
+        ResponseEntity<NodeTypeResponse> response = systemApiController.getNodeType();
```

Comment:
Should there be test for case where the getServerNodeType throws exception?

Reply:
The node type comes from a `.ini` file which is loaded into `SystemProperties` on startup. I think that it should be highly unlikely to get an exception from this method. It would probably mean that there were some errors loading the system properties in the first place, which would likely cause the system to not even start